src/Gaffer/BoxIO.cpp

```diff
@@ -244,6 +261,19 @@ void BoxIO::setupPromotedPlug()
 	Plug *toPromote = m_direction == Plug::In ? inPlugInternal() : outPlugInternal();
 	Plug *promoted = PlugAlgo::promoteWithName( toPromote, namePlug()->getValue() );
 	namePlug()->setValue( promoted->getName() );
+
+	if( m_direction == Plug::Out )
+	{
+		Box *box = parent<Box>();
+		BoolPlug *boxEnabledPlug = box->enabledPlug();
+		if( !boxEnabledPlug )
+		{
+			BoolPlugPtr p = new BoolPlug( g_enabledName, Plug::In, true, Plug::Default | Plug::Dynamic );
```

Comment:
Does this need some isExecuting protection, or does BoxIO::setup always get called after the enabled plug would have been added?